src/main/js/controls/Line/helpers/helpers.js

```diff
@@ -250,6 +250,77 @@ const draw = (scale, config, canvasSVG, dataTarget) => {
             .remove();
     }
 };
+
+/**
+ * Extracts regions inside value and prepare a list of value region
+ *
+ * @param {object} dataTarget - Data points object
+ * @param {Function} getXDataValues - checks if the value is valid and etract value for x-axis
+ *
+ * @returns {Array} value region subset
+ */
+const getValueRegionSubset = (dataTarget, getXDataValues) => {
+    const valueRegionSubset = [];
+    const valueRegions = [];
+    let valueRegion = {
+        color: undefined,
+        values: []
+    };
+    let previousColor;
+    dataTarget.values.forEach((value) => {
+        if (
+            !utils.isEmpty(value.region) &&
+            !utils.isEmpty(value.region.start) &&
+            !utils.isEmpty(value.region.end)
+        ) {
+            // If the color is different, then move it to new region set
+            if (previousColor !== value.region.color) {
```

Comment:
where will we get the value for previous color from? looks like it just declared in line 269

Reply:
previous color is just a temp storage of previously evaluated region. This condition I have added to check if there is any change in the color between the value regions.